File: python/arrays/binary_search.py
n = int(input("Enter size of sorted array: "))
arr = list(map(int, input(f"Enter {n} elements in ascending order: ").split()))
key = int(input("Enter element to search: "))

left, right = 0, n - 1
index = -1
while left <= right:
    mid = left + (right - left) // 2
    if arr[mid] == key:
        index = mid
        break
    elif arr[mid] < key:
        left = mid + 1
    else:
        right = mid - 1

if index != -1:
    print("Element found at index:", index)
else:
    print("Element not found in the array.")

File: C/arrays/binary_search.c
#include <stdio.h>
int main(){
    int n; scanf("%d",&n);
    int arr[n]; for(int i=0;i<n;i++) scanf("%d",&arr[i]);
    int key; scanf("%d",&key);
    int l=0,r=n-1,found=-1;
    while(l<=r){
        int mid=(l+r)/2;
        if(arr[mid]==key){ found=mid; break; }
        else if(arr[mid]<key) l=mid+1;
        else r=mid-1;
    }
    if(found!=-1) printf("Found at index %d",found); else printf("Not found");
}
